collector/zfs_linux.go

```diff
@@ -81,6 +98,50 @@ func (c *zfsCollector) parseProcfsFile(reader io.Reader, fmtExt string, handler
 	return scanner.Err()
 }
 
-func (c *zfsCollector) updatePoolStats(ch chan<- prometheus.Metric) (err error) {
+func (c *zfsCollector) parsePoolProcfsFiles(zpoolPaths []string, handler func(string, zfsSysctl, int)) (err error) {
+
+	for _, zpoolPath := range zpoolPaths {
+		file, err := os.Open(zpoolPath)
+		if err != nil {
+			log.Debugf("Cannot open %q for reading. Is the kernel module loaded?", zpoolPath)
+			err = zfsNotAvailableError
+		}
+		defer file.Close()
```

Comment:
Deferring in a loop is generally discouraged.  Since I imagine the number of ZFS pools on any given system should be pretty small, this may be okay, but it still feels a little funny.  May be worth refactoring out the scanning function to tighten up the loop body and then just explicitly close after the scanning function is done.